shared/util.go

```diff
@@ -268,49 +279,69 @@ func LogPath(path ...string) string {
 //     One of `file`, `symlink`, `directory`
 //   - `X-LXD-write`
 //     One of `overwrite`, `append`
-func ParseLXDFileHeaders(headers http.Header) (uid int64, gid int64, mode int, type_ string, write string) {
-	uid, err := strconv.ParseInt(headers.Get("X-LXD-uid"), 10, 32)
-	if err != nil {
-		uid = -1
-	}
+func ParseLXDFileHeaders(headers http.Header) (*LXDFileHeaders, error) {
+	var uid, gid int64 = -1, -1
+	var mode = -1
+	var err error
 
-	gid, err = strconv.ParseInt(headers.Get("X-LXD-gid"), 10, 32)
-	if err != nil {
-		gid = -1
+	rawUID := headers.Get("X-LXD-uid")
+	if rawUID != "" {
+		uid, err = strconv.ParseInt(rawUID, 10, 32)
+		if err != nil {
+			return nil, fmt.Errorf("Invalid UID: %w", err)
+		}
 	}
 
-	mode64, err := strconv.ParseInt(headers.Get("X-LXD-mode"), 10, 0)
```

Comment:
I can't for the life of me figure out why this was [done the way it was](https://github.com/canonical/lxd/commit/2eaad4ab40e9c527e06f2054a7aa937ed204edf7#diff-b69c194d748eb56e6d07348168e3dc9f65712eb80366a5dd7d6583518c8162faR110); the new behavior here is less strict (allows parsing `0b` or `0x` prefixed ints as well).